src/shogun/lib/DynamicArray.h

```diff
@@ -619,6 +604,16 @@ template <class T> class CDynamicArray :public CSGObject
 			m_array.resize_array(m_array.get_num_elements(), true);
 		}
 
+		CSGObject* clone()
+		{
+			CDynamicArray * cloned = (CDynamicArray*) CSGObject::clone();
+			// Since the array vector is registered with
+			// current_num_elements as size (see parameter
+			// registration) the cloned version has less memory
+			// allocated than known to dynarray. We fix this here.
+			cloned->m_array.num_elements = cloned->m_array.current_num_elements;
```

Comment:
So again: 
 * we don't register the size of allocated memory parameter
 * serializing/deserializing will change it. Storing only stores the used part of the array, loading only allocated as much as needed, and re-sizes when something is added next time
 * it is not part of equals as it is not registered